runtime/queries/metricsview_data_types.go

```diff
@@ -0,0 +1,171 @@
+package queries
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"io"
+	"strings"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+type MetricsViewDataTypes struct {
+	MetricsViewName    string                               `json:"metrics_view_name,omitempty"`
+	MetricsView        *runtimev1.MetricsViewSpec           `json:"-"`
+	ResolvedMVSecurity *runtime.ResolvedMetricsViewSecurity `json:"security"`
+
+	Result []*runtimev1.MetricsViewDataType `json:"-"`
+}
+
+var _ runtime.Query = &MetricsViewDataTypes{}
+
+func (q *MetricsViewDataTypes) Key() string {
+	return fmt.Sprintf("TableColumns:%s", q.MetricsViewName)
+}
+
+func (q *MetricsViewDataTypes) Deps() []*runtimev1.ResourceName {
+	return []*runtimev1.ResourceName{
+		{Kind: runtime.ResourceKindSource, Name: q.MetricsView.Table},
+		{Kind: runtime.ResourceKindModel, Name: q.MetricsView.Table},
+		{Kind: runtime.ResourceKindMetricsView, Name: q.MetricsViewName},
+	}
+}
```

Comment:
The deps should only return immediate dependencies. The responsibility to resolve the DAG should be an upstream concern (at the moment, this is handled by invalidating the cache based on `StateUpdatedOn`, which is increased in all cases where a ref changes).